docs(linux): U-Boot: Falcon-Mode: minor fixes for the 11.1 release (#359)
TexasInstruments/processor-sdk-doc: source/linux/Foundational_Components/U-Boot/UG-Falcon-Mode.rst

               |                               | CONFIG_SYS_LOAD_ADDR
       57MiB   |   Kernel + initramfs Image    | CONFIG_SPL_LOAD_FIT_ADDRESS
               |                               |
    0x85900000 +-------------------------------+
               |                               |
               |  R5 U-Boot SPL Stack + Heap   |
       39MiB   |       (size defined by        |
               | SPL_STACK_R_MALLOC_SIMPLE_LEN)|
               |                               |
    0x88000000 +-------------------------------+ CONFIG_SPL_STACK_R_ADDR
               |                               | K3_HW_CONFIG_BASE in ATF
       16MiB   |          Kernel DTB           | CONFIG_SPL_PAYLOAD_ARGS_ADDR
               |                               |
    0x89000000 +-------------------------------+
      331MiB   | Device Manager (DM) Load Addr |
    0x9db00000 +-------------------------------+
       12MiB   |          DM Reserved          |
    0x9e700000 +-------------------------------+
        1MiB   |            Unused             |
    0x9e800000 +-------------------------------+ BL32_BASE in ATF
       24MiB   |             OPTEE             |

>     0xa0000000 +===============================+ End of DDR (512MiB)

Reviewer: Ah, I missed this earlier. Out of alignment.

Author: Addressed in cd4e299a. Thanks for the review!

fitImage:
=========

The resulting :file:`fitImage` file in the boot directory of rootfs is produced
with the constituent binaries pre-signed with x509 certificates. This file is
authenticated from TIFS at boot time, which allows for a lower boot time than

> authenticating on the R5 core.

Reviewer: This seems like a duplication of the above paragraph. Similar restructuring can be used to quiet vale here. Also, I guess `rootfs` is a slang term. Realistically we should expand it to root filesystem for those unfamiliar with it.

Author: Thanks for pointing it out, I have addressed it in the latest commit, though I'm not sure why commit-check is failing...

Reviewer: Ah, that's because the correct tag would be `doc` not `docs`, but also as highlighted in the contribution guide that really should be reserved to actual project documentation https://github.com/TexasInstruments/processor-sdk-doc/blob/master/CONTRIBUTING.md#commit-formatting

Reviewer: Yeah, noticed general regression and fixed it upstream. Tracking locally in #379

=======

Falcon mode makes use of a cut down variant of the tispl binary called
:file:`tifalcon.bin` with the Cortex-A SPL and its corresponding DTB removed.
This file is deployed to the boot directory inside rootfs so it can be picked by
the R5 SPL at boot time.

R5 SPL:
=======

> The R5 SPL is used for loading the kernel :file:`fitImage` and :file:`tifalcon.bin`
> file, though the :file:`fitImage` for falcon boot is signed by using an x509
> certificate with TIFS keys instead of making use of signature nodes and keys
> present in the DT. This allows for faster authentication since TIFS uses the
> security accelerator for authentication, which is much faster than doing the
> same on R5 core.

Reviewer: ```suggestion
The R5 SPL loads the kernel :file:`fitImage` and :file:`tifalcon.bin` file. An
x509 certificate with TIFS keys verifies the :file:`fitImage` for falcon boot
instead of making use of signature nodes and keys present in the DT. This allows
for faster authentication since TIFS uses the security accelerator for
authentication, which is much faster than doing the same on R5 core.
```

Author: Change added to e2bfae0, thanks for the review!

> This support depends on the U-Boot's :file:`k3_r5_falcon.config` fragment, which is
> built alongside the standard R5 defconfig when ``ti-falcon`` is enabled. This

Reviewer: ```suggestion
This support is present alongside the standard R5 ``defconfig`` when ``ti-falcon``
is enabled due to U-Boot's :file:`k3_r5_falcon.config` fragment. This
```

Author: Added in e2bfae0.

   $ ./scripts/secure-binary-image.sh falcon.dtb falcon.dtb.sec
   $ mkimage -f fitImage.its fitImage

****************
Non-Yocto Users:
****************

> Follow the steps below to build ``tiboot3.bin``, ``tifalcon.bin`` and the

Reviewer 2: In this documentation, for highlighting file names and directories you can use role file

https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-file
```suggestion
Follow the steps below to build :file:`tiboot3.bin`, :file`tifalcon.bin` and the
```

Author: The final output appears the same in the formatted HTML, is there a reason for using :file: instead of just regular double backticks?

Reviewer 2: Great question! In Sphinx documentation, both the `:file:` role and regular double back-ticks can visually look similar in the rendered HTML, since both typically display mono-spaced text for filenames. However, there are important differences:

1. Semantic Meaning

> file: conveys to Sphinx (and readers) that the content is specifically a filename or file path.
Double backticks are generic and just mean “inline literal text,” without any semantic indication.

2. Extensions and Cross-References

>  Some Sphinx extensions can add features for :file: roles, like linking to files or tooltips.
Double backticks won’t be picked up by such extensions.

3. Consistency

> Using semantic roles makes your documentation more consistent and easier for contributors to follow established conventions. (this is currently not there but we are trying to slowly enforce it so that we do have it at some point of time)

So, use `:file:` when you want to semantically mark text as a file name or path, or when you want to benefit from possible custom styling, extensions, or improved accessibility. 

Use double back-ticks for generic inline code or literals as mentioned in sphinx docs - https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#inline-markup

Author: That makes sense, thanks for the explaination Chirag! I have added the changes in [5956ce2](https://github.com/TexasInstruments/processor-sdk-doc/pull/359/commits/5956ce2c3b074a073c8c16f09ab413b1e771894c).

   $ ./scripts/secure-binary-image.sh falcon.dtb falcon.dtb.sec
   $ mkimage -f fitImage.its fitImage

****************
Non-Yocto Users:
****************

> Follow the steps below to build ``tiboot3.bin``, ``tifalcon.bin`` and the

Reviewer 2: ```suggestion
Following are the steps to build ``tiboot3.bin``, ``tifalcon.bin`` and the
```

> [RedHat.TermsSuggestions] Depending on the context, consider using 'after', 'later', or 'following' rather than 'below'.
